lib/SILGen/SILGenLValue.cpp

```diff
@@ -843,8 +843,11 @@ namespace {
                          ManagedValue base) && override {
       assert(base.getType().isExistentialType() &&
              "base for open existential component must be an existential");
-      assert(base.getType().isAddress() &&
-             "base value of open-existential component was not an address?");
+      assert((base.getType().isAddress() ||
+              base.getType().getPreferredExistentialRepresentation(SGF.SGM.M) ==
+                  ExistentialRepresentation::Boxed) &&
+             "base value of open-existential component was not an address or a "
+             "boxed existential?");
```

Comment:
It's weird that boxed existentials end up in `OpenOpaqueExistentialComponent`, but I see that there's code there explicitly to handle them, so I guess it's as-intended.